components/button/button-split.js

```diff
@@ -0,0 +1,169 @@
+			/**
+			 * @ignore
+			 */
+			formaction: { type: String, reflect: true },
+			/**
+			 * @ignore
+			 */
+			formenctype: { type: String, reflect: true },
+			/**
+			 * @ignore
+			 */
+			formmethod: { type: String, reflect: true },
+			/**
+			 * @ignore
+			 */
+			formnovalidate: { type: String, reflect: true },
+			/**
+			 * @ignore
+			 */
+			formtarget: { type: String, reflect: true },
+			/**
+			 * @ignore
+			 */
+			name: { type: String, reflect: true },
+			/**
+			 * Styles the buttons as a primary buttons
+			 * @type {boolean}
+			 */
+			primary: { type: Boolean, reflect: true },
+			/**
+			 * ACCESSIBILITY: REQUIRED: Accessible text for the main action button
+			 * @type {string}
+			 */
+			text: { type: String, reflect: true, required: true },
+			/**
+			 * @ignore
+			 */
+			type: { type: String, reflect: true }
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				display: inline-block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.container {
+				display: flex;
+				gap: 6px;
+			}
+			.main-action {
+				--d2l-button-start-end-radius: 0;
+				--d2l-button-end-end-radius: 0;
+			}
+			.menu-opener {
+				--d2l-button-start-start-radius: 0;
+				--d2l-button-end-start-radius: 0;
+				--d2l-button-padding: 0 0.6rem;
+			}
+			.menu-opener[primary] > d2l-icon {
+				color: #ffffff;
+			}
+			::slotted(:not(d2l-menu)) {
+				display: none;
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		/** @ignore */
+		this.autofocus = false;
+		this.disabled = false;
+		this.primary = false;
+	}
+
+	static get focusElementSelector() {
+		return '.main-action';
+	}
+
+	render() {
+		return html`
+			<div class="container">
+				<d2l-button
+					?autofocus="${this.autofocus}"
+					class="main-action"
+					@click="${this.#handleMainActionClick}"
+					description="${ifDefined(this.description)}"
+					?disabled="${this.disabled}"
+					disabled-tooltip="${ifDefined(this.disabledTooltip)}"
+					form="${ifDefined(this.form)}"
+					formaction="${ifDefined(this.formaction)}"
+					formenctype="${ifDefined(this.formenctype)}"
+					formmethod="${ifDefined(this.formmethod)}"
+					?formnovalidate="${this.formnovalidate}"
+					formtarget="${ifDefined(this.formtarget)}"
+					name="${ifDefined(this.name)}"
+					?primary="${this.primary}"
+					type="${ifDefined(this.type)}">
+					${this.text}
+				</d2l-button>
+				<d2l-dropdown>
+					<d2l-button
+						aria-label="Other Actions"
+						class="d2l-dropdown-opener menu-opener"
+						?disabled="${this.disabled}"
+						?primary="${this.primary}">
+						<d2l-icon icon="tier1:chevron-down"></d2l-icon>
+					</d2l-button>
+					<d2l-dropdown-menu>
+						<slot name="menu"></slot>
+					</d2l-dropdown-menu>
+				</d2l-dropdown>
+			</div>
+		`;
+	}
+
+	#handleMainActionClick(e) {
+		e.stopPropagation();
+		/** Dispatched when the main action button is clicked */
+		this.dispatchEvent(new CustomEvent('d2l-button-split-click'));
```

Comment:
I'd also be content just dispatching a new `click` event, or only stopping propagation of the opener click if we want to retain the original composed path.

Reply:
Yeah I think generally we tend to prefer using native `click` and `change` events... otherwise as we've seen consumers will wire up to `click` anyway!

Reply:
I've updated to dispatch a click event with the key.